posts/2021-03-24-whats-new-in-MP-Rest-Client2.0.adoc

```diff
@@ -0,0 +1,212 @@
+author_github: https://github.com/andymc12
+seo-title: MicroProfile Rest Client 2.0 - First Look - OpenLiberty.io
+seo-description: This post describes the new features available in MicroProfile Rest Client 2.0 and how to use
+them.
+blog_description: "This post describes the new features available in MicroProfile Rest Client 2.0 and how to use
+them."
+---
+= MicroProfile Rest Client 2.0 - First Look
+Andy McCright <https://github.com/andymc12>
+
+The latest release of MicroProfile's type-safe REST client has a lot of new and exciting features. In this post, we'll
+take a look at some of the new features and how you can use them in Open Liberty, including:
+
+. <<query-param-style,Using QueryParamStyle to specify how collections of query parameters should be formatted>>
+. <<proxy-server,Proxy server support>>
+. <<auto-redirects,Automatically following redirects>>
+. <<sse-support,Support for Server Sent Events (SSEs)>>
+
+== Getting started
+
+The MP Rest Client 2.0 implementation is new in Open Liberty 21.0.0.3 - so make sure you are using the latest
+version of Liberty. Next, you will need to add the following link:{url-prefix}/guides/maven-intro.html[Maven] dependency
+to your pom.xml:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.eclipse.microprofile.rest.client</groupId>
+    <artifactId>microprofile-rest-client-api</artifactId>
+    <version>2.0</version>
+    <scope>provided</scope>
+</dependency>
+----
+
+or, if you use link:{url-prefix}/guides/gradle-intro.html[Gradle]:
+[source,gradle]
+----
+dependencies {
+    mpRestClient group: 'org.eclipse.microprofile.rest.client', name: 'microprofile-rest-client-api', version: '2.0'
+}
+----
+
+Also, make sure to configure your Liberty server with the `mpRestClient-2.0` feature in the server.xml:
+[source,xml]
+----
+<server>
+  <featureManager>
+    <feature>mpRestClient-2.0</feature>
+    <!-- ... -->
+  </featureManager>
+  <!-- ... -->
+</server>
+----
+
+That's it! Now that we've got our development and deployment environments set up, it's time to code!
+
+
+[#query-param-style]
+== Using QueryParamStyle to specify how collections of query parameters should be formatted
+
+Rest Client interfaces can specify query parameters using the `@QueryParam("paramName")` annotation, but often a server
+will require that multi-valued query parameters must be formatted in a certain way. For example, suppose we have a
+client interface like:
+[source,java]
+----
+@RegisterRestClient
+public interface MyClient {
+    @GET
+    String multiValues(@QueryParam("myParam") List<String> values);
+}
+----
+By default, if a caller invokes the the client with `multiValues(Arrays.asList("a", "b", "c"))` the MP Rest Client will
+produce an HTTP request with multiple key/value pairs - something like: `?myParam=a&myParam=b&myParam=c`
+
+Most servers will handle that just fine, but other servers will require that the HTTP request be a single key with a 
```

Comment:
suggested edit

Most servers will handle that just fine, but other servers will require that the HTTP request be a single key ...

--->

Although most servers will handle that just fine, some servers might require the HTTP request to be a single key...